src/home/homework-luks.c

```diff
@@ -1670,12 +1670,16 @@ static struct crypt_pbkdf_type* build_good_pbkdf(struct crypt_pbkdf_type *buffer
         assert(buffer);
         assert(hr);
 
+        bool benchmark = user_record_luks_pbkdf_force_iterations(hr) == UINT32_MAX;
+
         *buffer = (struct crypt_pbkdf_type) {
                 .hash = user_record_luks_pbkdf_hash_algorithm(hr),
                 .type = user_record_luks_pbkdf_type(hr),
-                .time_ms = user_record_luks_pbkdf_time_cost_usec(hr) / USEC_PER_MSEC,
+                .time_ms = benchmark ? user_record_luks_pbkdf_time_cost_usec(hr) / USEC_PER_MSEC : 0,
+                .iterations = benchmark ? 0 : user_record_luks_pbkdf_force_iterations(hr),
```

Comment:
so this doesn't look right, i.e. with this what happens if i set the field to 0? and what if it is set to UINT32_MAX?

I presume a "0" that doesn't mean "benchmark" cannot actually be specified, right?

So here's what I think we should do:

1. Use UINT64_MAX for the indicator for "use benchmarked value", not UINT32_MAX
2. use CLAMP() to clamp all other values to the range 1…UINT32_MAX (which appears to be the valid range for libcryptsetup)